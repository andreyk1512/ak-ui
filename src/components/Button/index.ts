export * from './Button';
export * from './Button.types';

import { Link } from './components';
import { Button as OriginButton } from './Button';

export type ButtonProps = typeof OriginButton & {
  Link: typeof Link;
};

export const Button = OriginButton as ButtonProps;
Button.Link = Link;
