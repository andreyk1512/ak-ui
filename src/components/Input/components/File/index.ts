export * from './File';
export * from './File.types';
