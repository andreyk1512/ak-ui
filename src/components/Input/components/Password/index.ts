export * from './Password';
export * from './Password.types';
