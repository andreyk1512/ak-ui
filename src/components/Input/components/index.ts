export * from './Password';
export * from './TextArea';
export * from './File';
