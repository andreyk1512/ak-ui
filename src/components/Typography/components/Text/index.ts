export * from './Text.tsx';
export * from './Text.types';
