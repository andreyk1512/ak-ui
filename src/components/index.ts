export * from './Tag';
export * from './Button';
export * from './Checkbox';
