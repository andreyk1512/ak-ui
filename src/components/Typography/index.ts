export * from './Typography';
export * from './Typography.types';

import { Text, Title, Paragraph } from './components';
import { Typography as OriginTypography } from './Typography';

export type TypographyProps = typeof OriginTypography & {
  Text: typeof Text;
  Title: typeof Title;
  Paragraph: typeof Paragraph;
};

export const Typography = OriginTypography as TypographyProps;
Typography.Text = Text;
Typography.Title = Title;
Typography.Paragraph = Paragraph;
