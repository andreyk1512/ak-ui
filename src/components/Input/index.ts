export * from './Input';
export * from './Input.types';

import { Password, TextArea, File } from './components';
import { Input as OriginInput } from './Input';

export type InputProps = typeof OriginInput & {
  Password: typeof Password;
  TextArea: typeof TextArea;
  File: typeof File;
};

export const Input = OriginInput as InputProps;
Input.Password = Password;
Input.TextArea = TextArea;
Input.File = File;
