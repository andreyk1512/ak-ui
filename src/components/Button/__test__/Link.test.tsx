import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Link } from '../components';

describe('Button.Link', () => {
  const name = 'Link';
  const href = 'https://link.to/';

  it('should render the link', () => {
    const { container } = render(<Link href={href}>{name}</Link>);
    const link = screen.getByRole('link', { name });

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', href);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('shouldn\'t be able to click on disabled link', () => {
    render(<Link href={href}>{name}</Link>);
    const link = screen.getByRole('link', { name });

    expect(link).toHaveAttribute('href', undefined);
  });
});
