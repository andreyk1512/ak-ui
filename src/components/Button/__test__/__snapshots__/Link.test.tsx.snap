// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.Link should render the link 1`] = `
<a
  class="css-575uon"
  href="https://link.to/"
>
  Link
</a>
`;
